File: week06/32/241009.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        result = 0
        q = deque()
        q.append(root)

        while q:
            cur = q.popleft()
            if (cur.left and not cur.left.left and not cur.left.right):
                result += cur.left.val
            if cur.left:
                q.append(cur.left)

            if cur.right:
                q.append(cur.right)

        return result

File: week06/32/241009.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        int result=0;
        queue<TreeNode*> q;

        q.push(root);

        while (!q.empty()){
            TreeNode * cur = q.front();
            q.pop();
            if (cur->left && cur->left->left == NULL && cur->left->right==NULL) {
                result += cur->left->val;
            }
            if (cur->left) {
                q.push(cur->left);
            } 
            if (cur->right) {
                q.push(cur->right);
            }
        }
        return result;
    }
};